spec/controllers/api/v0/order_cycles_controller_spec.rb

```diff
@@ -70,6 +70,26 @@ module Api
           expect(product_ids).to include product1.id, product2.id
           expect(product_ids).to_not include product3.id
         end
+
+        context "with supplier properties" do
+          let!(:supplier_property) { create(:property, presentation: 'Certified Organic') }
+          let!(:supplier) { create(:supplier_enterprise, properties: [supplier_property]) }
+
+          before do
+            product1.update!(supplier: supplier)
+            product2.update!(supplier: supplier)
+            product3.update!(supplier: supplier)
+            product4.update!(supplier: supplier, inherits_properties: false)
+          end
+
+          it "filter out the product that don't inherits from supplier properties" do
+            api_get :products, id: order_cycle.id, distributor: distributor.id,
+                               q: { properties_id_or_supplier_properties_id_in_any: [supplier_property.id] }
```

Comment:
This filter is the problem. It does say to return any products that match the supplier properties. So when applying this filter, the current result is correct. We just need to look at the product's properties. The difficulty is that the product's properties need to include the supplier properties if the inherit flag is set.

I'm not sure how to best implement this. If we want to solve it in SQL, we need to join the supplier properties only if the flag is set. Maybe we can call that column inherited_properties and then the filter parameter would be this:
```suggestion
                               q: { properties_id_or_inherited_properties_id_in_any: [supplier_property.id] }
```
You need to create that additional join for this though.